src/Expander.hs

```diff
@@ -1072,26 +1074,25 @@ addRootScope' stx = do
   return (addScope' stx rsc)
 
 
-forkExpandOneDecl :: DeclPtr -> Scope -> DeclValidityPtr -> Syntax -> Expand ()
-forkExpandOneDecl dest sc phaseDest stx = do
-  forkExpanderTask $ ExpandDecl dest sc stx phaseDest
+forkExpandOneDecl :: DeclTreePtr -> Scope -> DeclValidityPtr -> Syntax -> Expand ()
+forkExpandOneDecl dest sc declValidityPtr stx = do
+  forkExpanderTask $ ExpandDeclTree dest sc stx declValidityPtr
 
-forkExpandDecls :: ModBodyPtr -> Syntax -> Expand ()
+forkExpandDecls :: DeclTreePtr -> Syntax -> Expand ()
 forkExpandDecls dest (Syntax (Stx _ _ (List []))) =
-  modifyState $ over expanderCompletedModBody (<> Map.singleton dest Done)
+  linkDeclTree dest DeclTreeLeaf
 forkExpandDecls dest (Syntax (Stx scs loc (List (d:ds)))) = do
   -- Create a scope for this new declaration
   sc <- freshScope $ T.pack $ "For declaration at " ++ shortShow (stxLoc d)
```

Comment:
I'm doing this one step at a time, and for now I'm still generating the scope inside `forkExpandDecls`. Next, I want to move this line to each `declPrim`.

Reply:
I have now removed this `freshScope` call and replaced it with multiple `freshScope` calls elsewhere, one per `primDecl`.